modules/nextflow/src/main/groovy/nextflow/scm/ProviderConfig.groovy

```diff
@@ -111,8 +111,8 @@ class ProviderConfig {
         if( p != -1 )
             result = result.substring(p+3)
         p = result.indexOf('/')
-        if( p != -1 )
-            result = result.substring(0,p)
+        if( p > 0 )
+            result = result.substring(0, p)
         return result
```

Comment:
I think the logic should be a more explicit ie. if it contains `://` it does the same as now, otherwise it should start with `/` ?

Also add a test here 

https://github.com/seqeralabs/nextflow/blob/ae04fc374cafb822af844d3b3479f2d52f6b7c40/modules/nextflow/src/test/groovy/nextflow/scm/ProviderConfigTest.groovy#L72-L100

Reply:
About tests. they are [here](https://github.com/nextflow-io/nextflow/blob/local-repo/modules/nextflow/src/test/groovy/nextflow/scm/ProviderConfigTest.groovy#L104:L120) already

Reply:
About the logic, in case of a local repository can the domain start with `file://`? If that's the case, we have to handle this too and making it explicitly might maybe end up being too verbose...

Reply:
Don't remember this, but it could make sense to handle it.

Reply:
An possiility would be:
```groovy
def result = server ?: path
def p = result.indexOf('://')
if( p != -1 )
    result = result.substring(p+3)
if ( result.startsWith('/') )
    return result
p = result.indexOf('/')
if( p != -1 )
    result = result.substring(0, p)
return result
```

Reply:
Cool, please add some comment to explain the mess behind e.g. remote url, local url, abs paths. Tx!